presto-verifier/src/main/java/com/facebook/presto/verifier/checksum/ArrayColumnChecksum.java

```diff
@@ -63,19 +78,29 @@ public boolean equals(Object obj)
         }
         ArrayColumnChecksum o = (ArrayColumnChecksum) obj;
         return Objects.equals(checksum, o.checksum) &&
+                Objects.equals(floatingPointChecksum, o.floatingPointChecksum) &&
                 Objects.equals(cardinalityChecksum, o.cardinalityChecksum) &&
                 Objects.equals(cardinalitySum, o.cardinalitySum);
     }
 
     @Override
     public int hashCode()
     {
-        return Objects.hash(checksum, cardinalityChecksum, cardinalitySum);
+        return Objects.hash(checksum, floatingPointChecksum, cardinalityChecksum, cardinalitySum);
     }
 
     @Override
     public String toString()
     {
-        return format("checksum: %s, cardinality_checksum: %s, cardinality_sum: %s", checksum, cardinalityChecksum, cardinalitySum);
+        if (!floatingPointChecksum.isPresent()) {
```

Comment:
nit: instead of repeating everything for constructing the string, you can use MoreObject.toStringHelper to construct a nicely printed string with all the fields. Something like:
```
MoreObjects.ToStringHelper toStringHelper = MoreObjects.toStringHelper(ArrayColumnchecksum)
    .add("cardinality_checksum", cardinalityChecksum)
    .add("cardinality_sum", cardinalitySum)

if(floatingPointChecksum.isPresent() {
    toStringHelper.add("floating_point_checksum", floatingPointchecksum.get());
} 
else {
    toStringHelper.add("checksum", checksum);
}

return toStringHelper.toString();
```

Reply:
not sure if it's important that this toString matches the toStrings of the other checksums, so I'll add the caveat that the generated string will look slightly different than what you have, see the docs for example output https://guava.dev/releases/19.0/api/docs/com/google/common/base/MoreObjects.html#toStringHelper(java.lang.Object)

Reply:
@rschlussel 
I don't know. I don't feel that this change worth the trouble at the moment.
You are saying "to construct a nicely printed string with all the fields" - are they going to be more nicely printed than with the current code? How?

Maybe we can consider a small refactor followup to move all Checksums to such format?

Reply:
not necessarily nicer than the current code. just helps you with not duplicating for the floating point vs. non-floating point case.  It would look something like `ArrayColumnChecksum{cardinality_checksum=123, cardinality_sum=345, ...}`. anyway, fine to leave it as is.